src/Event/TransformEvent.php

```diff
@@ -78,15 +79,15 @@ public function getFields()
     }
 
     /**
-     * @return mixed
+     * @return object
      */
     public function getObject()
     {
         return $this->object;
     }
 
     /**
-     * @param mixed $document
+     * @param Document $document
      */
     public function setDocument($document)
```

Comment:
Note that I have not added the typehint here, as it would be a BC break (method is not final)